agent/Retriever.py

```diff
@@ -0,0 +1,76 @@
+from llama_index import (
+    VectorStoreIndex,
+    SummaryIndex,
+    SimpleKeywordTableIndex,
+    SimpleDirectoryReader,
+    ServiceContext,
+)
+from llama_index.agent import OpenAIAgent
+from llama_index.schema import IndexNode
+from llama_index.tools import QueryEngineTool, ToolMetadata
+from llama_index.llms import OpenAI
+import os
+from llama_index.retrievers import RecursiveRetriever
+from llama_index.query_engine import RetrieverQueryEngine
+from llama_index.response_synthesizers import get_response_synthesizer
+
+class Retriever:
+    def __init__(self, index_path):
+        
+        self.llm = OpenAI(temperature=0, model="gpt-4-1106-preview")
+        self.service_context = ServiceContext.from_defaults(llm=self.llm)
+    
+    def load_documents(self, files):
+        self.docs = {}
+        for file in files.keys():
+            docs[file] = SimpleDirectoryReader(
```

Comment:
There is a variable scoping issue: `docs[file]` should be `self.docs[file]`.
```suggestion
            self.docs[file] = SimpleDirectoryReader(
```